src/PantheonTerminalWindow.vala

```diff
@@ -788,6 +802,97 @@ namespace PantheonTerminal {
             return (TerminalWidget)((Gtk.Bin)tab.page).get_child ();
         }
 
+        uint name_check_timeout_id = 0;
+        private void schedule_name_check () {
+            if (name_check_timeout_id > 0) {
+                Source.remove (name_check_timeout_id);
+            }
+
+            name_check_timeout_id = Timeout.add (50, () => {
+                if (!check_for_tabs_with_same_name ()) {
+                    return true;
+                } else {
+                    name_check_timeout_id = 0;
+                    return false;
+                }
+            });
+        }
+
+        /** Compare every tab label with every other and resolve ambiguities **/
+        private bool check_for_tabs_with_same_name () {
+            /* Take list copies so foreach clauses can be nested safely*/
+            var terms = terminals.copy ();
+            var terms2 = terminals.copy ();
+
+            foreach (TerminalWidget t in terms) {
```

Comment:
Personally, I find it easier to read (& write) complex code with short variable names otherwise I cannot see the wood for the trees.  However, I can easily replace the short variables with longer names in keeping with the preferred style.